spec.bs

```diff
@@ -1717,3 +1764,147 @@ specification is printed below:
+
+  When not empty, the <{iframe/allow}> attribute will result in adding an allowlist for each
+  recognized feature to the iframe element’s content navigable's container policy or the fencedframe
+  element's [=fenced navigable container/fenced navigable=]'s container policy, when it is
+  constructed.
+</div>
+
+<div algorithm=create-permissions-policy-response>
+  Modify the definition of [=Create a Permissions Policy for a navigable from response=] to read:
+
+  Given null, a [=navigable container=], or a [=fenced navigable container=] (|container|), an
+  [=origin=] (|origin|), a [=response=] (<var ignore>response</var>), and null or a
+  [=list=] of [=features=] (|required features|), this algorithm returns a new
+  [=Document/permissions policy=].
+
+  Modify step 1 of the algorithm to read:
+  1. Let <var ignore>policy</var> be the result of running
+    [=Create a Permissions Policy for a navigable=] given |container|, |origin|, and
+    |required features|.
+</div>
+
+<div algorithm=shared-document-creation-changes>
+  Modify the [=create and initialize a Document object=] algorithm. Rewrite step 3 to read:
+
+  3. Let <var ignore>permissionsPolicy</var> be the result of [=Create a Permissions Policy for a
+    navigable from response|creating a permissions policy from a response=] given
+    |navigationParams|'s [=navigable=]'s [=navigable container|container=], |navigationParams|'s
+    [=navigation params/origin=], |navigationParams|'s [=navigation params/response=], and
+    |navigationParams|'s [=navigable=]'s [=fenced frame config instance=]'s [=fenced frame config
+    instance/required permissions to load=].
+</div>
+
+<div algorithm=new-browsing-context-changes>
+  Modify the [=create a new browsing context and document=] algorithm. Rewrite step 7 to read:
+
+  7. Let <var ignore>permissionsPolicy</var> be the result of [=Create a Permissions Policy for a
+    navigable|creating a permissions policy=] given <var ignore>embedder</var>,
+    <var ignore>origin</var>, and null.
+
+  Note: This change is made in addition to the changes to [=create a new browsing context and
+  document=] outlined in [[#creating-browsing-contexts-patch]].
+</div>
+
+<div algorithm=attempt-populate-history-patches>
+  Modify [[HTML]]'s [=attempt to populate the history entry's document=] algorithm. Add a step
+  before the step inside the [=queue a task|queued task=] starting with "If
+  |failure| is true, then:" that reads:
+
+  8. Otherwise, if the result of [=Should navigation response to navigation request be blocked by
+    Permissions Policy?=] given |navigationParams|'s [=request=],
+    |navigationParams|'s [=response=], and <var ignore>navigable</var> is "`Blocked`", then set
+    |failure| to true.
+</div>
+
+<div algorithm=permissions-policy-block-request>
+  Create a new algorithm called <dfn>Should navigation response to navigation request be blocked by
+  Permissions Policy?</dfn> in [[!HTML]].
```

Comment:
Is the intention of this algorithm really to not consider any of the permissions policies in the response? My understanding is that we wanted to make sure that the required permissions were not blocked by either:
 1. A parent document's permissions that were inherited in the disabled state
 2. Response headers in the actual response

Do we not care about (2) above? Is the only piece of the response that we need, the origin?

Reply:
Re: response permissions policies, [we have a WPT](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/fenced-frame/default-enabled-features-attribute-allow.https.html;l=25-35;drc=3ccc9b67e688ec5f98f6664385f0587fb724d307) to check that we can further restrict permissions after a fenced frame loads

Re: the wildcard, [our implementation](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/navigation_request.cc;l=8367-8384;drc=dfb15f3bc92332c4f433ad9721ba1a16ce2595cc) matches what's in the spec.
`the policy must be either explicitly enabled for 'origin', or the policy must by default be enabled for all origins`